### ISSUE TITLE
Batching / bulk queries - MaxBatchSize()

### ISSUE THREAD

**Issue by blixmo** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#151]
### Steps to reproduce
	using (var dataContext = GetDataContext())
	{
		await dataContext.Database.EnsureCreatedAsync();
		
		var account = new Account { Name = "Account 1") };
		var account2 = new Account { Name = "Account 2") };
		
		dataContext.Accounts.Add(account);
		dataContext.Accounts.Add(account2);
		
		dataContext.SaveChanges();
	}

	private static ApplicationDbContext GetDataContext()
	{
		const string connectionString = "myConnectionString";

		var optionsBuilder = new DbContextOptionsBuilder<ApplicationDbContext>();
		optionsBuilder.UseMySql(connectionString, options => options.MaxBatchSize(2));
		return new ApplicationDbContext(optionsBuilder.Options);
	}
	
	public class ApplicationDbContext : BaseDbContext
    {
        public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : base(options) { }
	}
	
	public class BaseDbContext
    {
        // Just DbSets, entities etc.
	}
	
	public class Account
    {
        public int Id { get; set; }
	
        public string Name { get; set; }
    }

### The issue
If you specify the MaxBatchSize as an option for the DbContextOptionsBuilder the SaveChanges()-method will fail. Everything works fine if I remove the MaxBatchSize() option or if I specify it to MaxBatchSize(1). But as long as input number is > 1 it won't work.

```
Exception message: Database operation expected to affect 2 row(s) but actually affected 1 row(s). Data may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=527962 for information on understanding and handling optimistic concurrency exceptions.

Stack trace: at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ThrowAggregateUpdateConcurrencyException(Int32 commandIndex, Int32 expectedRowsAffected, Int32 rowsAffected)
   at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ConsumeResultSetWithPropagation(Int32 commandIndex, DbDataReader reader)
   at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.Consume(DbDataReader reader)
   at Microsoft.EntityFrameworkCore.Update.Internal.MySqlModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Update.Internal.MySqlBatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IReadOnlyList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Notified.Tests.DatabaseTests.<SetupAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Notified.Tests.DatabaseTests.<Test>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass46_1.<<InvokeTestMethodAsync>b__1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.ExecutionTimer.<AggregateAsync>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xunit.Sdk.ExceptionAggregator.<RunAsync>d__9.MoveNext()
```

**Comment by caleblloyd** [comment]
I'm looking into this.  Exception is coming from [AffectedCountModificationCommandBatch.cs](https://github.com/aspnet/EntityFramework/blob/dev/src/Microsoft.EntityFrameworkCore.Relational/Update/AffectedCountModificationCommandBatch.cs).  I'm able to recreate, should have a fix soon.

**Comment by blixmo** [comment]
- Removes bulk insert logic. We can't use it because we must call LAST_INSERT_ID() after each insert to get generated IDs. I don't think there's any other way to do this with MySQL.

This won't solve the issue since it's the bulk inserts I need... The only reason to use the MaxBatchSize() is to activate the bulk operations, including inserts. Are you sure there is no way to retrive the generated IDs from a bulk insert with MySQL?

Maybe it should be possible to use the bulk insert anway and just don't return the generated IDs? My case is that I've a list of x tousands items and it's so ugly (and stupid) to insert all these items one by one to the database instead of all at once in one batch. There is no need to get the generated IDs back in this case.

**Comment by caleblloyd** [comment]
Shipped in [1.1.1-prerelease-10004](https://www.nuget.org/packages/Pomelo.EntityFrameworkCore.MySql/1.1.1-prerelease-10004)
